projects/packages/my-jetpack/_inc/components/product-cards-section/index.tsx

```diff
@@ -90,17 +92,48 @@ const DisplayItems: FC< DisplayItemsProps > = ( { slugs } ) => {
 
 interface ProductCardsSectionProps {
 	noticeMessage: ReactNode;
+	siteIsRegistered?: boolean;
 }
 
-const ProductCardsSection: FC< ProductCardsSectionProps > = ( { noticeMessage } ) => {
-	const { ownedProducts = [], unownedProducts = [] } =
+const ProductCardsSection: FC< ProductCardsSectionProps > = ( {
+	noticeMessage,
+	siteIsRegistered,
+} ) => {
+	const { ownedProducts: initialOwnedProducts = [], unownedProducts: initialUnownedProducts = [] } =
 		getMyJetpackWindowInitialState( 'lifecycleStats' );
+	const [ productsOwnership, setProductsOwnership ] = useValueStore( 'productsOwnership', {
+		ownedProducts: initialOwnedProducts,
+		unownedProducts: initialUnownedProducts,
+	} );
+
+	const {
+		data: productOwnershipData,
+		isLoading,
+		refetch: refetchOwnershipData,
+	} = useProductsByOwnership();
+
+	useEffect( () => {
+		if ( ! isLoading ) {
+			if ( siteIsRegistered ) {
+				refetchOwnershipData();
+			}
+
+			const { ownedProducts = [], unownedProducts = [] } = productOwnershipData;
+			setProductsOwnership( { ownedProducts, unownedProducts } );
+		}
```

Comment:
If possible, let's avoid nesting conditions. In this case, for example, we can just mention:
```
if ( isLoading ) {
  return;
}